sql/catalyst/src/main/java/org/apache/spark/sql/vectorized/ArrowColumnVector.java

```diff
@@ -130,9 +130,16 @@ public ColumnarMap getMap(int rowId) {
   @Override
   public ArrowColumnVector getChild(int ordinal) { return childColumns[ordinal]; }
 
+  ArrowColumnVector(DataType type) {
```

Comment:
So is the idea to call this constructor and then override `initAccessor` to provide a custom accessor?
If so, would it be easier to just add a constructor `public ArrowColumnVector(ArrowVectorAccessor accessor)` with the relaxed access modifiers?

Reply:
Thanks. My intention was to allow sub classes to override the constructor so that the subclass constructor can do more stuff before calling `initAccessor`. At least that is my use case.
The subclasses can choose to override `initAccessor` or the subclass constructor can directly assign the `accessor`.

Is your idea for `public ArrowColumnVector(ArrowVectorAccessor accessor)` more for using ArrowColumnVector as is rather than extending?

Reply:
Sounds reasonable to me.